File: gemini/nivel basico/python/suma.py
def sumar_numeros(a, b):
  """
  Esta función recibe dos números y devuelve su suma.
  """
  return a + b

try:
  num1 = float(input("Ingresa el primer número: "))
  num2 = float(input("Ingresa el segundo número: "))

  resultado = sumar_numeros(num1, num2)
  print(f"La suma de {num1} y {num2} es: {resultado}")
except ValueError:
  print("Error: Por favor, ingresa números válidos.")

File: gemini/nivel basico/go/suma.go
package main

import (
	"fmt"
	"strconv"
	"bufio"
	"os"
	"strings"
)

func sumarNumeros(a float64, b float64) float64 {
	return a + b
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Print("Ingresa el primer número: ")
	num1Str, _ := reader.ReadString('\n')
	num1Str = strings.TrimSpace(num1Str) // Eliminar espacios y saltos de línea

	fmt.Print("Ingresa el segundo número: ")
	num2Str, _ := reader.ReadString('\n')
	num2Str = strings.TrimSpace(num2Str)

	num1, err1 := strconv.ParseFloat(num1Str, 64)
	num2, err2 := strconv.ParseFloat(num2Str, 64)

	if err1 != nil || err2 != nil {
		fmt.Println("Error: Por favor, ingresa números válidos.")
		return
	}

	resultado := sumarNumeros(num1, num2)
	fmt.Printf("La suma de %.2f y %.2f es: %.2f\n", num1, num2, resultado)
}